# HG changeset patch
# User Jordi Gutiérrez Hermoso <jordigh@gmail.com>
# Date 1298478845 21600
#      Wed Feb 23 10:34:05 2011 -0600
# Node ID 95b1c64c287f751eab796659e8a11d5b727d59c9
# Parent  c40528d06c9285e4f0b35e8a13a34d6e03b16068
Improve .hgignore adding more things to be ignored

diff --git a/.hgignore b/.hgignore
--- a/.hgignore
+++ b/.hgignore
@@ -1,8 +1,39 @@
-(^|/)CVS($|/)
-(^|/)\.hg($|/)
-(^|/)\.hgtags($|/)
+syntax: regexp
+# The recurrent (^|/) idiom in the regexps below should be understood
+# to mean "at any directory" while the ^ idiom means "from the
+# project's top-level directory".
+
+# This one gets created by gnulib during the build, contains no Octave
+# files.
+^gnulib/
+
+# These directories mostly contain cruft during build time, but they
+# do contain some Octave code.
+^libgnu/
+^m4/
+
+# Emacs tools create these
+(^|/)TAGS$
+(^|/)semantic.cache$
 
+(/|^)Makefile\.in$
+^INSTALL$
+
+^aclocal\.m4
 ^build-.*($|/)
-(^|/)configure$
-(^|/)autom4te.cache($|/)
+^configure$
+^autom4te\.cache($|/)
 ^config\.h\.in$
+
+# e.g. doc/faq/OctaveFAQ.info
+#      doc/interpreter/octave.info-4
+^doc/.*\.info(-\d)?$
+
+^doc/\w*/stamp-vti$
+^doc/\w*/version\.texi$
+^doc/interpreter/images\.mk$
+
+# e.g. liboctave/smx-op-inc.mk
+^liboctave/[\w-]*\.mk$
+
+^src/DLD-FUNCTIONS/module\.mk$
diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2011-02-23  Jordi Gutiérrez Hermoso  <jordigh@gmail.com>
+
+	* .hgignore: Cleanup and more files to be ignore that get added
+	to the source during build time.
+
 2011-02-22  Rik  <octave@nomad.inbox5.com>
 
 	* Makefile.am: New explicit 'make check' target.
 
 2011-02-21  John W. Eaton  <jwe@octave.org>
 
 	* NEWS: Note that PCRE is now required to build Octave.
 	* configure.ac: Improve check for PCRE, which is now required.
